examples/animation/animated-character.html

```diff
@@ -0,0 +1,256 @@
+
+        // Set the canvas to fill the window and automatically change resolution to be the same as the canvas size
+        app.setCanvasFillMode(pc.FILLMODE_FILL_WINDOW);
+        app.setCanvasResolution(pc.RESOLUTION_AUTO);
+
+        window.addEventListener("resize", function () {
+            app.resizeCanvas(canvas.width, canvas.height);
+        });
+
+        // Create an Entity with a camera component
+        var cameraEntity = new pc.Entity();
+        cameraEntity.addComponent("camera", {
+            clearColor: new pc.Color(0, 0, 0)
+        });
+        cameraEntity.translateLocal(2.5, 3.5, 2.5);
+        cameraEntity.lookAt(0, 0, 0);
+
+        // Create light in the scene
+        app.scene.ambientLight = new pc.Color(1.0, 1.0, 1.0);
+        var lightEntity = new pc.Entity();
+        lightEntity.name = 'light';
+        lightEntity.addComponent("light", {
+            type: "directional",
+            color: new pc.Color(1.0, 1.0, 1.0),
+            intensity: 0.5,
+            range: 20
+        });
+        lightEntity.setLocalEulerAngles(45, 0, -45);
+
+        // Create boxes which will signal character movement
+        var box1 = new pc.Entity();
+        box1.addComponent('model', {
+            type: 'box'
+        });
+        box1.setLocalPosition(1, 0.5, 0);
+        box1.setLocalScale(0.5, 0.5, 0.5);
+        var box2 = new pc.Entity();
+        box2.addComponent('model', {
+            type: 'box'
+        });
+        box2.setLocalPosition(-1, 0.25, -4);
+        box2.setLocalScale(0.25, 0.25, 0.25);
+
+        app.assets.loadFromUrl(
+            "../assets/textures/playcanvas-grey.png",
+            "texture",
+            function (err, asset) {
+                var material = new pc.StandardMaterial();
+                material.diffuseMap = asset.resource;
+                material.update();
+                box1.model.meshInstances[0].material = material;
+                box2.model.meshInstances[0].material = material;
+            }
+        );
+
+        // Create a plane for the character to walk on
+        var planeEntity = new pc.Entity();
+        planeEntity.addComponent("model", {
+            type: "plane"
+        });
+        planeEntity.setLocalScale(4, 1, 30);
+        planeEntity.setPosition(0, 0, 0);
+        var material = new pc.StandardMaterial();
+        material.diffuse = new pc.Color(0.15, 0.20, 0.3);
+        material.update();
+        planeEntity.model.meshInstances[0].material = material;
+
+        // Create the character model entity
+        var modelEntity = new pc.Entity();
+        modelEntity.name = 'model';
+
+        app.root.addChild(box1);
+        app.root.addChild(box2);
+        app.root.addChild(planeEntity);
+        app.root.addChild(cameraEntity);
+        app.root.addChild(lightEntity);
+        app.root.addChild(modelEntity);
+
+        app.assets.loadFromUrl('../assets/modelStateGraphLarge.json', 'animationstategraph', function (err, stateGraph) {
```

Comment:
This code might be neater if you load the assets from an array of filenames in a loop instead of the nesting.